packages/core/src/shared/sam/build.ts

```diff
@@ -293,3 +311,34 @@ export async function resolveBuildFlags(buildFlags: string[], samCliVersion: Sem
     }
     return buildFlags
 }
+
+/**
+ * Returns true if there's an ongoing build process for the provided template, false otherwise
+ * @Param templatePath The path to the template.yaml file
+ */
+function isBuildInProgress(templatePath: string): boolean {
+    const expirationDate = getRecentResponse(buildProcessMementoRootKey, globalIdentifier, templatePath)
+    if (expirationDate) {
+        return Date.now() < parseInt(expirationDate)
+    }
+    return false
+}
+
+/**
+ * Throws an error if there's a build in progress for the provided template
+ * @Param templatePath The path to the template.yaml file
+ */
+export function throwIfTemplateIsBeingBuilt(templatePath: string) {
```

Comment:
for the following functions does burying them in another function give us anything of value? If not they can live in the top level

Reply:
I extracted those functions because the runBuild() function was getting too big. I think having them outside makes runBuild() easier to read.